### PR TITLE
teleport-kube-agent-app: Secure Kubernetes access via Teleport

### PR DESCRIPTION
**Pull request by tuladhar**
Towards https://github.com/giantswarm/roadmap/issues/2494
### Checklist

- [x] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/test create
/test upgrade
/run cluster-test-suites

### CODE REVIEW

File: helm/default-apps-aws/values.yaml [hunk 1]
```diff
@@ -202,6 +202,29 @@ apps:
...
+    # a list of extraConfigs for the App,
+    # It can be secret or configmap
+    # https://github.com/giantswarm/rfc/tree/main/multi-layer-app-config#example
+    extraConfigs:
+      - kind: secret
+        name: "{{ $.Values.clusterName }}-teleport-join-token"
+        namespace: "{{ $.Release.Namespace }}"
+      - kind: configMap
+        name: "{{ $.Values.clusterName }}-teleport-kube-agent-config"
+        namespace: "{{ $.Release.Namespace }}"
```
Comment (calvix) [inline helm/default-apps-aws/values.yaml:227]: as this will block the app deployment until the resource is avaiable

**Review by calvix — CHANGES_REQUESTED** [review]
should be only merged once teleport-operator is deployed and working in all CAPA MCs

File: helm/default-apps-aws/values.yaml (hunk 1 above)
Comment (tuladhar) [reply helm/default-apps-aws/values.yaml:227]: Shouldn't be a problem cause teleport-operator will be creating those as soon as Cluster CR is reconciled, before default apps is deployed.

**Comment by tuladhar** [comment]
> should be only merged once teleport-operator is deployed and working in all CAPA MCs

Thanks for the heads up 👍

File: helm/default-apps-aws/values.yaml [hunk 2]
```diff
@@ -202,6 +202,26 @@ apps:
...
+    catalog: default-test
+    clusterValues:
+      configMap: true
+      secret: true
+    enabled: true
+    forceUpgrade: true
+    namespace: kube-system
+    # used by renovate
+    # repo: giantswarm/teleport-kube-agent-app
+    version: 0.4.0-f06a6d1a33515030b2c7cbd98d10d9f5bdcbe2cb
```
Comment (anvddriesch) [inline helm/default-apps-aws/values.yaml:217]: All of this looks good to me but let's only merge it after testing thorougly and releasing teleport :)

**Comment by tuladhar** [comment]
> should be only merged once teleport-operator is deployed and working in all CAPA MCs

teleport-operator has been deployed on all CAPA MCs.

**Review by calvix — APPROVED** [review]

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**